[BE] 히스토리 조회 및 삭제 API 구현

Ojeegu opened the pull request:
## What is this PR? 👓
- 활동 기록 목록 조회 시 `is_deleted = false`인 기록들을 반환하는 API 구현
  - 관련 테스트 코드 작성
- 활동 기록 목록 삭제하는 API 구현 
  - 예외 관련 테스트 코드를 제외한 테스트 코드 작성  

## Key changes 🔑
- DB의 created_at 시간 값을 String 타입으로 변환하기 위해 `HistoryFindAllResponse` 클래스에 `calculateElapsedTime` 메서드 작성
- `deletedByIds`의 파라미터로 List<Long> 타입 전달
- 목록 삭제 시 파라미터로 전달 받은 id 개수와 전달 받은 id 중 DB에 존재하면서 `is_deleted = false`인 id의 개수를 비교
  -  유효하지 않은 id가 있는 경우 불일치하므로 예외 발생

## To reviewers 👋
- 목록 삭제 기능에서 유효하지 않은 id 요청 시 `RuntimeException` 예외를 발생 시켰습니다.
  - 추후 예외 발생(Json 타입 반환) 수정과 함께 예외 관련 테스트 코드도 작성할 예정입니다.

## 테스트 실행 결과
![image](https://github.com/codesquad-team-06/todo-max/assets/107015624/5b935616-0ae9-4b9a-8546-e0b592e2c678)

## `/histories` API 요청 결과
![image](https://github.com/codesquad-team-06/todo-max/assets/107015624/f05cb4b0-e88a-4414-b3c5-ba2c40c86cc4)

## Issues
Closes #5

Ojeegu committed:
#5 🎨 style : 접근 제어자 및 줄바꿈 변경

- Builder class 접근제어자 default -> public 변경
- 코딩 컨벤션에 맞게 줄바꿈 수정

Ojeegu committed:
#5 ✨ feat : 활동 목록 불러오기 및 전체 삭제 기능 추가

Ojeegu committed:
#5 ✅ test : 유효한 활동 목록을 불러오는 Test 코드 작성

Ojeegu committed:
#5 ✅ test : 활동 기록을 삭제하는 Test 코드 작성

Ojeegu committed:
#5 🎨 style : 메서드명 변경 및 줄바꿈 변경

Ojeegu committed:
#5 ✅ test : 활동 기록 조회 후 Json으로 반환하는 Test 코드 작성

Ojeegu committed:
#5 ♻️ refactor : validate 메서드 반환 타입 및 예외 발생 위치 변경

- validate 메서드의 반환 타입을 void -> boolean 변경
- validate에서 발생시켰던 예외를 deleteByIds로 이동

Ojeegu committed:
#5 ✅ test : 활동 목록 삭제 Test 코드 작성

Ojeegu committed:
#5 🎨 style : 불필요한 class 삭제 및 코딩 컨벤션 적용

yonghwankim-dev commented on be/src/main/java/codesquad/todo/history/controller/HistoryRestController.java:
```diff
@@ -1,15 +1,35 @@
 	}
+
+	@GetMapping
+	public ResponseEntity<List<HistoryFindAllResponse>> getHistories() {
```
```suggestion
	public List<HistoryFindAllResponse> getHistories() {
```
위와 같이 변경하여도 동일한 결과가 나올것입니다.

Ojeegu replied:
반영했습니다. 감사합니다.

yonghwankim-dev reviewed (approved):
고생하셨습니다. 제가 보기엔 문제 없는 것 같습니다.!
그리고 PR에 적어주신 /histories API 요청 결과에서 prevColumn과 nextColumn은 컬럼 아이디번호가 아닌 컬럼 제목이어야 합니다. 아무래도 샘플 데이터를 잘못된 것을 사용하신 것 같습니다. ERD 페이지의 샘플 데이터를 참고해주시면 감사하겠습니다.

Ojeegu committed:
#5 ♻️ refactor : getHistories() 반환 타입 변경

- 반환 타입을 ResponseEntity<List<HistoryFindAllResponse>> -> List<HistoryFindAllResponse> 로 변경
- 같은 기능을 하므로 코드 최소화

swinb reviewed (approved):
고생하셨습니다!!

jsjseop reviewed (approved):
👍

Ojeegu committed:
Merge branch 'be' into feature#5
